pkg/schema/account_type.go

```diff
@@ -0,0 +1,84 @@
+package schema
+
+// AccountType represents the type of an account in the system, in the format of a string that displays it's qualifiers
+// in the format of {ROLE}.{PLATFORM}.{STORAGE_METHOD}. For example, "HOST.STELLAR.ENV" represents a host account
+// that is used in the Stellar platform and stored in the environment.
+type AccountType string
+
+const (
+	HostStellarEnv                    AccountType = "HOST.STELLAR.ENV"
+	ChannelAccountStellarDB           AccountType = "CHANNEL_ACCOUNT.STELLAR.DB"
+	DistributionAccountStellarEnv     AccountType = "DISTRIBUTION_ACCOUNT.STELLAR.ENV"      // was "ENV_STELLAR"
+	DistributionAccountStellarDBVault AccountType = "DISTRIBUTION_ACCOUNT.STELLAR.DB_VAULT" // was "DB_VAULT_STELLAR"
+	DistributionAccountCircleDBVault  AccountType = "DISTRIBUTION_ACCOUNT.CIRCLE.DB_VAULT"  // was "DB_VAULT_CIRCLE"
+)
```

Comment:
👋 👋 👋  This is the main change in this PR! All other files are collaterals of this change.

It's about having a more generic representation of `TransactionAccount` that can be used for all these cases.

I decided to represent the triad `{ROLE}.{PLATFORM}.{STORAGE_METHOD}` in a single string because this way we have a more manageable set of options. If we had three parameters to represent these types, we'd need to interpolate them and figure out which ones were supported or unsupported on the fly, which would be more complicated and error-prone.